### PR TITLE
Proposal of refactoring

### PR DESCRIPTION
**Pull request by gaawa**
# Proposal of refactoring

Even with the small code base, it got rather unreadable.
Therefore, a proposal for refactoring was created.

* Using a more OOP-approach, a class was created for contents in bot.py
* Two config classes was created in config.py
    * JsonConfig contains values read from json configuration file
    * ObjConfig contains some values from JsonConfig and others, converted to discord.py objects
* entrance_handler.py contains EntranceHandler class that is responsible for entrance-system related processes
* member_comparator.py contains SameMemberComparator and possibly in the future, DifferentMemberComparator. The thought behind this is, since for example, on_member_update method will require comparisons on numbers of member object members, a class is created that contains all comparison methods.

### CODE REVIEW

File: desugreet/member_comparator.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class SameMemberComparator:
```
Comment (TypicalFence) [inline desugreet/member_comparator.py]: couldn't this class just be 2 functions? we always temporarily create it and it doesn not manage state really, it's just `input -> output`

File: desugreet/entrance_handler.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import discord
+
+class EntranceHandler:
```
Comment (TypicalFence) [inline desugreet/entrance_handler.py]: can we rename it to `EntranceService`?

I feel like a `Handler` should have the same methods as the the bot provides such as `on_message` etc.

Comment (gaawa) [reply desugreet/entrance_handler.py]: I'll rename it

File: desugreet/member_comparator.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    def __init__(self, memberBefore, memberAfter) -> None:
+        self.memberBefore = memberBefore
+        self.memberAfter = memberAfter
+
+    def roleWasAdded(self, role):
```
Comment (TypicalFence) [inline desugreet/member_comparator.py]:
```suggestion
    def role_was_added(self, role):
```

**Comment by TypicalFence** [comment]
small thing: we should try to stick to one way of spelling variables, python generally uses `snake_case` but `PascalCase` for class names

File: desugreet/member_comparator.py (hunk 1 above)
Comment (gaawa) [reply desugreet/member_comparator.py]: will be changed as discussed

File: desugreet/member_comparator.py (hunk 3 above)
Comment (gaawa) [reply desugreet/member_comparator.py]: was actually unintended

**Pull request merged**